packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -667,6 +671,22 @@ See documentation: ${createDocumentationLink({
     this.middleware.forEach(({ instance }) => {
       instance.started();
     });
+
+    // This is the automatic Insights middleware,
+    // added when `insights` is unset and the initial results possess `queryID`.
+    // Any user-provided middleware will be added later and override this one.
+    if (typeof this._insights === 'undefined') {
+      mainHelper.derivedHelpers[0].once('result', () => {
+        const hasQueryID = this.mainIndex
+          .getScopedResults()
```

Comment:
does this also work when no root indexName is given? i believe no result happens then, but to check. If it doesn't call the callback in that case, we need to listen to all derivedHelpers, not just the first one.

(it reminds me that i'm not 100% sure whether https://github.com/algolia/instantsearch/blob/bed6c5fd003da7ba85618cd5f2fb4ff1e60c50f9/packages/instantsearch.js/src/lib/server.ts#L17-L19 works correctly too)